File: bubble sort/bubble.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

using namespace std;

void bubbleSort(int arr[], int n) {
    for (int i = 0; i < n - 1; i++)
        for (int j = 0; j < n - i - 1; j++)
            if (arr[j] > arr[j + 1])
                swap(arr[j], arr[j + 1]);
}

int main() {
    const int SIZE = 20;
    int arr[SIZE];

    srand(time(0));
    for (int i = 0; i < SIZE; i++)
        arr[i] = rand() % 100 + 1;

    cout << "Arreglo original:\n";
    for (int i = 0; i < SIZE; i++)
        cout << arr[i] << " ";
    cout << endl;

    bubbleSort(arr, SIZE);

    cout << "\nArreglo ordenado:\n";
    for (int i = 0; i < SIZE; i++)
        cout << arr[i] << " ";
    cout << endl;

    return 0;
}

File: bubble sort/bubble.py
import random
arreglo = [random.randint(1, 100) for _ in range(20)]

print("Arreglo original:")
print(arreglo)

def bubble_sort(lista):
    n = len(lista)
    for i in range(n):
        for j in range(0, n - i - 1):
            if lista[j] > lista[j + 1]:
                lista[j], lista[j + 1] = lista[j + 1], lista[j]

bubble_sort(arreglo)
print("\nArreglo ordenado con Bubble Sort:")
print(arreglo)